File: CoreJava/SOLID-princples-app/src/srp/refractor/InvoicePrinter.java
package srp.refractor;

public class InvoicePrinter {
	public void printInvoice(Invoice invoice) {
		System.out.println("Invoice ID \t\t: " + invoice.getId());
		System.out.println("Customer name \t\t: " + invoice.getName());
		System.out.println("Cost Price \t\t: " + invoice.getCost());
		System.out.println("Discounted Price \t: " + invoice.calcDiscount());
		System.out.println("Tax on this product \t: " + invoice.calcTax());
		System.out.println("Total \t\t\t: " + invoice.calcTotal());
	}
}

File: DotNET/Solid Principles/SRPRefractorApp/SRPRefractorApp/InvoicePrinter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace SRPRefractorApp
{
    class InvoicePrinter
    {
        public void printInvoice(Invoice invoice)
        {
            Console.WriteLine("Invoice ID :" + invoice.Id);
            Console.WriteLine("Customer name :" + invoice.Name);
            Console.WriteLine("Cost Price :" + invoice.Cost);
            Console.WriteLine("Discounted Price :" + invoice.CalculateDiscount());
            Console.WriteLine("Tax on this product :" + invoice.CalculateTax());
            Console.WriteLine("Total :" + invoice.CalculateTotal());
        }
    }
}
